nix/process-compose/default.nix

```diff
@@ -45,17 +45,21 @@ in
           text = ''
             ${preHook}
 
-            run-process-compose () {
-              set -x; process-compose ${cliOutputs.global} --config ${configFile} "$@"; set +x
-            }
 
-            # Run `up` command, with arguments; unless the user wants to pass their own subcommand.
-            if [ "$#" -eq 0 ]; then
-              run-process-compose up ${cliOutputs.up}
-            else
-              run-process-compose "$@"
+            # If there are no arguments, it's the "up" command
+            # If the first argument is "up", it's also the "up" command
+            # If the first argument starts with a dash, we assume there isn't a subcommand, so it's also the "up" command
+            # Otherwise, we assume it's a subcommand other than "up"
+            params=(${cliOutputs.global})
+            set +u
+            if [ -z "$1" ] || [[ "$1" == "up" ]] || [[ "$1" == -* ]] ; then
+              params+=(--config ${configFile} ${cliOutputs.up})
```

Comment:
In the `"$1" == "up"` case, shouldn't you drop the `up` from `$@`? Otherwise, in line 60 it will be passed twice.

Reply:
No, we're passing the arguments on as we receive them, and we never add an "up" ourselves afaik

Reply:
You are right:

```
❮ nix run --override-input process-compose-flake .. .#default -- up --log-file=output.log
[...]
+ process-compose --no-server --config /nix/store/bl46rll1p24g5fbdypbxsk6rfnklfp2z-process-compose-default.json --disable-dotenv --theme Cobalt up --log-file=output.log
```

But then, I just noticed something new here. Note that `--disable-dotenv --theme Cobalt` (defined in `cli.up`) appears **before** the `up` argument. It looks like their CLI parser accepts this out of order arguments anyway. But for clarity, I'd prefer that `--disable-dotenv --theme Cobalt` appear **after** the `up` argument, because those options are meant for `up` subcommand anyway.

Wait, I was wrong. Actually those are global options! We should fix this in options.
<img width="490" alt="image" src="https://github.com/user-attachments/assets/5d3b321d-91b0-43d9-a88e-7d34a61ee18d">

---

`--sort` however is an `up` specific option, so that's a better illustration:

```diff
--- a/example/flake.nix
+++ b/example/flake.nix
@@ -30,8 +30,7 @@
               };
               # Options for `process-compose up`
               up = {
-                disable-dotenv = true;
-                theme = "Cobalt";
+                sort = "health";
               };
             };
             settings = {
```

Without arguments, we see:

```
❯ nix run --override-input process-compose-flake .. .#default
[..]
+ process-compose --no-server --config /nix/store/bl46rll1p24g5fbdypbxsk6rfnklfp2z-process-compose-default.json --sort health
```

(And I'd still like to make `up` implicit here)

With `up` argument:

```
❯ nix run --override-input process-compose-flake .. .#default up
+ process-compose --no-server --config /nix/store/bl46rll1p24g5fbdypbxsk6rfnklfp2z-process-compose-default.json --sort health up
```

Here, I think `--sort health` should appear **after** `up` to make it clear (to the user who happens to diagnose our script output) that these flags are for `up` command only.

Reply:
I agree, it's better to put user arguments first. I've changed that.